File: models/product.js
const mongoose = require('mongoose')

const productSchema = new mongoose.Schema({
    _id: {
        type: mongoose.Schema.Types.ObjectId
    },
    name: {
        type: String,
        required: true
    },
    barcode: {
        type: String,
        required: true,
    },
    purchase_price: {
        type: Number,
    },
    sale_price: {
        type: Number,
    },
    seige_code: {
        type: String,
    },
    supplier: {
        type: String,
    },
    brand: {
        type: String,
    }
},
    { timestamps: true }
)

module.exports = mongoose.model('Product', productSchema)

File: lib/models/product.dart
import 'package:json_annotation/json_annotation.dart';

part 'product.g.dart';

@JsonSerializable()
class Product{

  @JsonKey(name: '_id')
  String? id;

  String name;
  String barcode;
  String seige_code;
  String supplier;
  String brand;
  int? purchase_price;
  int? sale_price;


  Product({required this.name, this.id, required this.barcode,required this.seige_code,required this.supplier,
    required this.brand,required this.purchase_price,required this.sale_price});

  factory Product.fromJson(Map<String, dynamic> json) => _$ProductFromJson(json);

  Map<String, dynamic> toJson() => _$ProductToJson(this);
}
